docs/failure-detector/failure-detector.md

```diff
@@ -0,0 +1,50 @@
+## _Failure detector_
+
+Main component in the failure detector is RemoteMonitoringService.
+The scheduler starts RemoteMonitoringService every 3 seconds (by default).
+
+Failure detector component (RemoteMonitoringService) consists of 4 parts: 
+ - collect poll report
+ - correct wrong epochs
+ - healing process
+ - failure detector
+ 
+### Collect poll report
+ - Each node:
+      - pings other nodes 3 times and build local NodeState. 
+      - collects wrong epochs.
+      - collects remote NodeSate's from other nodes.
+ - Aggregate ClusterState from the previous step.     
+
+### Wrong epoch detector
+ - if there is at least one wrong epoch in the cluster, then trigger the correction process. 
+ - check that the slot is unfilled. If the slot is unfilled then fill the slot.
+ - if there was at least one `wrong epoch` response then finish current iteration of a failure detection
+
+### Healing process
+ - Build a ClusterGraph based on the aggregated ClusterState (the cluster state built on the poll report stage)
+ - check if local node is in the unresponsive list.
+ - figure out if we can heal our self:
+   1. the node have to be fully connected which means it have to be connected to all responsive nodes.
```

Comment:
nit: ...which means it has* to be connected to..

Reply:
done